src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -601,9 +615,10 @@ int wm_vuldet_compare_pkg(cve_vuln_pkg *Pkg1, cve_vuln_pkg *Pkg2) {
         return -1;
     }
 
-    if (!(!Pkg1->arch && !Pkg2->arch) &&
-        !(Pkg1->arch && Pkg2->arch &&
-        !strcmp(Pkg1->arch, Pkg2->arch))) { // Architecture
+    if (!Pkg1->arch || !Pkg2->arch) { // Architecture
```

Comment:
This condition evaluates as `true` then, for example, we have the architecture of package 1 but not for package 2 (and the opposite) but, what happens with the situation where `!Pkg1->arch && !Pkg2->arch`? Are we considering the packages as different when we don't have both architectures?

Reply:
That `if` was made as a simple error handler, so there's no logic behind it. That being said, your assumption is true, although an empty architecture is rare since the OVAL function [handles](https://github.com/wazuh/wazuh/blob/331151a0c46aedca201f9574e7d76422c94af77d/src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c#L1646-L1648) that case. Not so with the NVD function, but perhaps we could do the same [here](https://github.com/wazuh/wazuh/blob/331151a0c46aedca201f9574e7d76422c94af77d/src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c#L1643-L1645). Doing so we could get rid of that `if` statement and make the code more readable. Any thoughts about it?

Reply:
Considering your explanation and that the OVAL function handles that case, I would leave the condition as it is. Thanks for the explanation.